indexer/services/ender/__tests__/lib/kafka-publisher.test.ts

```diff
@@ -139,11 +169,247 @@ describe('kafka-publisher', () => {
+        },
+        {
+          ...subaccount,
+          eventIndex: subaccount.eventIndex + 1,
+        },
+      ],
+    ])('successfully subaccounts events by %s', (
+      _field: string,
+      beforeSubaccount: AnnotatedSubaccountMessage,
+      afterSubaccount: AnnotatedSubaccountMessage,
+    ) => {
+      const publisher: KafkaPublisher = new KafkaPublisher();
+      const consolidatedBeforeSubaccount:
+      ConsolidatedKafkaEvent = createConsolidatedKafkaEventFromSubaccount(
+        beforeSubaccount,
+      );
+      const consolidatedAfterSubaccount:
+      ConsolidatedKafkaEvent = createConsolidatedKafkaEventFromSubaccount(
+        afterSubaccount,
+      );
+
+      publisher.addEvents([
+        consolidatedAfterSubaccount,
+        consolidatedSubaccount,
+        consolidatedBeforeSubaccount,
+      ]);
+
+      publisher.sortEvents(publisher.subaccountMessages);
+      expect(publisher.subaccountMessages).toEqual([beforeSubaccount, subaccount, afterSubaccount]);
+    });
+  });
+
+  describe('aggregateFillEventsForSubaccountMessages', () => {
+    const fill: FillFromDatabase = {
+      id: FillTable.uuid(testConstants.defaultTendermintEventId, Liquidity.TAKER),
+      subaccountId: testConstants.defaultSubaccountId,
+      side: OrderSide.BUY,
+      liquidity: Liquidity.TAKER,
+      type: FillType.MARKET,
+      clobPairId: '1',
+      orderId: testConstants.defaultOrderId,
+      size: '10',
+      price: '20000',
+      quoteAmount: '200000',
+      eventId: testConstants.defaultTendermintEventId,
+      transactionHash: '', // TODO: Add a real transaction Hash
+      createdAt: testConstants.createdDateTime.toISO(),
+      createdAtHeight: testConstants.createdHeight,
+      clientMetadata: '0',
+      fee: '1.1',
+    };
+    const order: OrderFromDatabase = {
+      ...testConstants.defaultOrderGoodTilBlockTime,
+      id: testConstants.defaultOrderId,
+    };
+
+    const recipientSubaccountId: IndexerSubaccountId = IndexerSubaccountId.fromPartial({
+      owner: 'recipient',
+      number: 1,
+    });
+    const deposit: TransferFromDatabase = {
+      id: '',
+      senderWalletAddress: defaultWalletAddress,
+      recipientSubaccountId: SubaccountTable.uuid(
+        recipientSubaccountId.owner,
+        recipientSubaccountId.number,
+      ),
+      assetId: testConstants.defaultAsset.id,
+      size: '10',
+      eventId: testConstants.defaultTendermintEventId,
+      transactionHash: 'hash',
+      createdAt: DateTime.utc().toISO(),
+      createdAtHeight: '1',
+    };
+    it('successfully aggregates all fill events per order id and sorts messages', async () => {
+      const publisher: KafkaPublisher = new KafkaPublisher();
+
+      // merged with message 3.
+      const msg1Contents: SubaccountMessageContents = {
+        fills: [
+          generateFillSubaccountMessage(fill, 'BTC-USD'),
+        ],
+      };
+      const message1: AnnotatedSubaccountMessage = {
```

Comment:
nit: Would it be possible to use message1 as a base to construct message 2/3, e.g.
```
{
  ...message1,
  eventIndex: 2,
  etc.
}
```

Reply:
done